observabilityclient/v1/rbac.py

```diff
@@ -0,0 +1,121 @@
+#   Copyright 2023 Red Hat, Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+from observabilityclient.utils.metric_utils import format_labels
+from keystoneauth1.exceptions.auth_plugins import MissingAuthPlugin
+import re
+
+
+class ObservabilityRbacError(Exception):
+    pass
+
+
+class Rbac():
+    def __init__(self, client, session, disable_rbac=False):
+        self.client = client
+        self.session = session
+        self.disable_rbac = disable_rbac
+        try:
+            self.project_id = self.session.get_project_id()
+            self.default_labels = {
+                    "project": self.project_id
+                    }
+            self.rbac_init_successful = True
+        except MissingAuthPlugin:
+            self.project_id = None
+            self.default_labels = {
+                    "project": "no-project"
+                    }
+            self.rbac_init_successful = False
+
+    def _enrich_labels(self, labels, disable_rbac):
+        if not self.rbac_init_successful and not disable_rbac:
+            raise ObservabilityRbacError("Unauthorized. Couldn't "
+                                         "get project id")
+        if not disable_rbac and not self.disable_rbac:
+            # Include labels to enforce rbac
+            if labels is not None:
+                labels = {**labels, **self.default_labels}
+            else:
+                labels = self.default_labels
+        return labels
+
+    # TODO aren't the additional labels just making
+    #      the code confusing? Are they useful?
```

Comment:
My 2cents: Maybe and No. User can construct whatever query with various labels in it, so I personally don't see a point of having additional parameters for additional parameters.

Reply:
I agree, I removed the code